modules/server/src/main/scala/scaladex/server/route/ProjectPages.scala

```diff
@@ -111,6 +111,35 @@ class ProjectPages(env: Env, database: WebDatabase, searchEngine: SearchEngine)(
           }
         }
       },
+      get {
+        path(projectM / "artifacts") { ref =>
+          getProjectOrRedirect(ref, user) { project =>
+            val latestArtifactsF = database.getLatestArtifacts(ref, preferStableVersions = false)
+            val headerF = service.getProjectHeader(project)
```

Comment:
The project header already contains the list of latest artifacts. So you don't need to run `getLatestArtifacts` a second time.

You can do:

```scala
val latestArtifacts = header.toSeq.flatMap(_.artifacts)
```